NTupleMaker/interface/functionsCP.h

```diff
@@ -263,6 +233,33 @@ TLorentzVector chargedPivec(const AC1B * analysisTree, int tauIndex){
 };
 
 int chargedPiIndex(const AC1B * analysisTree, int tauIndex){
+
+  int ncomponents = analysisTree->tau_constituents_count[tauIndex];
+  int piIndex=-1;
+  float maxPt=-1;
+  int sign = -1;
+  if(analysisTree->tau_charge[tauIndex]>0) sign = 1; 
+  for(int i=0;i<ncomponents;i++){ //selects the highest energy Pi with the same sign of the tau. 
+    //    cout<<"analysisTree->tau_constituents_pdgId[tauIndex][i] "<<analysisTree->tau_constituents_pdgId[tauIndex][i]<<endl;
+
+if((analysisTree->tau_constituents_pdgId[tauIndex][i]*sign)==211){
+//    if((analysisTree->tau_constituents_pdgId[tauIndex][i]*sign)==211||(analysisTree->tau_constituents_pdgId[tauIndex][i]*sign)==-13||(analysisTree->tau_constituents_pdgId[tauIndex][i]*sign)==-11){
+      TLorentzVector lvector; lvector.SetXYZT(analysisTree->tau_constituents_px[tauIndex][i],
+					      analysisTree->tau_constituents_py[tauIndex][i],
+					      analysisTree->tau_constituents_pz[tauIndex][i],
+					      analysisTree->tau_constituents_e[tauIndex][i]);
+      double Pt = lvector.Pt();
+      if(Pt>maxPt){
+	piIndex=i;
+	maxPt = Pt;
+      }
+    }
+  }
+  
+  return piIndex;
+};
+
+int chargedMuIndex(const AC1B * analysisTree, int tauIndex){
```

Comment:
Given that the lepton coming from the tau in mt and et channels is stored in a different collection compared to the tau constituents, can you please explain me how this function is used?

Reply:
Hi Andrea,
the function is obsolete, and can be removed ;-) It was for some vague other ambition..

Reply:
Ok, perfect. I will proceed to remove it once the PR is approved